torch/utils/_pytree.py

```diff
@@ -752,7 +752,7 @@ def tree_map_(func: Callable[..., Any], tree: PyTree, *rests: PyTree) -> PyTree:
     """
     leaves, treespec = tree_flatten(tree)
     flat_args = [leaves] + [treespec.flatten_up_to(r) for r in rests]
-    deque(map(func, *flat_args), maxlen=0)  # consume and exhaust the iterable
+    tuple(map(func, *flat_args))  # consume and exhaust the iterable
```

Comment:
This isn't exactly equivalent and will be slower: https://stackoverflow.com/questions/36763084/how-to-efficiently-exhaust-an-iterator-in-a-oneliner

Reply:
It's OK, we don't care too much about the performance of `tree_map_` right now